src/indexer/segment_writer.rs

```diff
@@ -103,11 +103,15 @@ impl SegmentWriter {
     /// be used afterwards.
     pub fn finalize(mut self) -> crate::Result<Vec<u64>> {
         self.fieldnorms_writer.fill_up_to_max_doc(self.max_doc);
-        let mapping = if let Some(settings) = self.segment_serializer.segment().index().settings() {
-            Some(sort_index(settings.clone(), &mut self)?)
-        } else {
-            None
-        };
+        let mapping = self
```

Comment:
Can you explicitly add the type here? Without editor hinting, it is like a riddle :D